vm/src/obj/objstr.rs

```diff
@@ -429,6 +430,29 @@ impl PyString {
         }
     }
 
+    #[pymethod(name = "__mod__")]
+    fn modulo(&self, values: PyObjectRef, vm: &VirtualMachine) -> PyResult {
+        let format_string_text = &self.value;
+
+        match CFormatString::from_str(format_string_text.as_str()) {
```

Comment:
the `.as_str()` suffix should not be required, since `&str` is a good type which is often compatible.